test/staking/SimpleDAIStaking.test.ts

```diff
@@ -532,257 +566,203 @@ contract("SimpleDAIStaking - staking with DAI mocks", ([founder, staker]) => {
 
-    await dai.approve(simpleStaking.address, approvedAmount, {
-      from: staker
-    });
+    await dai
+      .connect(staker)
+      .approve(goodCompoundStaking.address, approvedAmount);
 
-    const error = await simpleStaking
-      .stake(approvedAmount, {
-        from: staker
-      })
+    const error = await goodCompoundStaking
+      .connect(staker)
+      .stake(approvedAmount)
       .catch(e => e);
 
     expect(error.message).not.to.be.empty;
   });
 
   it("should emit a DAIStaked event", async () => {
-    const weiAmount = web3.utils.toWei("100", "ether");
-    await dai.mint(staker, weiAmount);
-    await dai.approve(simpleStaking.address, weiAmount, {
-      from: staker
-    });
+    let weiAmount = ethers.utils.parseEther("100");
+    await dai["mint(address,uint256)"](staker.address, weiAmount);
+    await dai.approve(goodCompoundStaking.address, weiAmount);
 
-    const transaction = await simpleStaking
-      .stake(weiAmount, 100, {
-        from: staker
-      })
-      .catch(console.log);
-
-    assert(transaction.logs[1].event === "Staked");
-    assert.equal(transaction.logs[1].args.value.valueOf(), weiAmount);
+    const transaction = await (
+      await goodCompoundStaking.connect(staker).stake(weiAmount, 100)
+    ).wait();
 
-    await simpleStaking.withdrawStake(weiAmount, {
-      from: staker
-    });
-  });
+    expect(transaction.events.find(_ => _.event === "Staked")).not.to.be.empty;
+    expect(
+      transaction.events.find(_ => _.event === "Staked").args.value.toString()
+    ).to.be.equal(weiAmount.toString());
 
-  it("should not withdraw interest to owner if cDAI value is lower than the staked", async () => {
-    const weiAmount = web3.utils.toWei("1000", "ether");
-    await dai.mint(staker, weiAmount);
-    await dai.approve(simpleStakingFM.address, weiAmount, {
-      from: staker
-    });
-    await simpleStakingFM
-      .stake(weiAmount, 100, {
-        from: staker
-      })
-      .catch(console.log);
-    const gains = await simpleStakingFM.currentUBIInterest();
-    const cdaiGains = gains["0"];
-    const precisionLossDai = gains["2"].toString(); //last 10 decimals since cdai is only 8 decimals while dai is 18
-    const fundBalanceBefore = await cDAI.balanceOf(founder);
-    await evm_mine(BLOCK_INTERVAL);
-    await simpleStakingFM.setFundManager(founder);
-    const res = await simpleStakingFM.collectUBIInterest(founder);
-    await simpleStakingFM.setFundManager(fundManager.address);
-    const fundBalanceAfter = await cDAI.balanceOf(founder);
-    expect(cdaiGains.toString()).to.be.equal("0");
-    expect(precisionLossDai.toString()).to.be.equal("0");
-    expect(fundBalanceAfter.toString()).to.be.equal(fundBalanceBefore.toString());
-    await simpleStakingFM.withdrawStake(weiAmount, {
-      from: staker
-    });
-  });
-
-  it("should not be able to stake if the getting an error while minting new cdai", async () => {
```

Comment:
why?

Reply:
Because we cant get it fail according to our code right now with cDAIMock